src/firecracker/src/main.rs

```diff
@@ -128,6 +130,16 @@ fn main() {
                 .takes_value(false)
                 .requires("config-file")
                 .help("Optional parameter which allows starting and using a microVM without an active API socket.")
+        )
+        .arg(
+            ArgInfo::new("log")
+                .takes_value(true)
+                .requires("metrics")
+        )
+        .arg(
+            ArgInfo::new("metrics")
+                .takes_value(true)
+                .requires("log")
```

Comment:
I agree, but the `LoggerConfig` structure that is used for deserialization requires both fifos to be set. Maybe we should make them optional instead or just the metrics one.

Reply:
I'd be more incline towards an approach that decouples the metrics and log. As it stands now, throughout the code the dependency between these two seems a bit artificial.

Reply:
I strongly advise looking at the `refactoring/epoll_handler` branch before moving ahead with this PR.

The Logger and Metrics have been mostly separated there, also neither is owned by vmm anymore.

I suggest developing the functionality in this PR directly on that branch or wait for it to be merged in master. Otherwise, I fear this might be throwaway work.